src/alire/alire-index_on_disk.adb

```diff
@@ -131,4 +131,43 @@ package body Alire.Index_On_Disk is
       return Hash (Hash'First .. Hash'First + Unique_Id_Length - 1);
    end Unique_Id;
 
+   ------------
+   -- Verify --
+   ------------
+
+   function Verify (This : Index'Class) return Outcome is
+      use GNAT.OS_Lib;
+   begin
+      if not Is_Directory (This.Metadata_Directory) then
+         return Outcome_Failure ("Metadata folder over index not found");
+      elsif not Is_Directory (This.Index_Directory) then
+         return Outcome_Failure ("Index folder not found");
+      else
+         --  TODO: simply try to load it once the loaded index is not global
+         --  For now, instead, locate index version file:
```

Comment:
Agreed, I removed the comment. I think that some later refactor already took care of the duplication that you mention, since I cannot find it right now.